### PR TITLE
feature/LPFG-187-authentication-resolve

### PR DESCRIPTION
**Pull request by superafroman**
Note this includes LPFG-188 (revoke) as well.

### CODE REVIEW

File: src/main/java/uk/gov/cshr/service/security/ClientDetails.java [hunk 1]
```diff
@@ -20,6 +20,7 @@
 
     private static final Set<String> GRANT_TYPES = new HashSet<String>() {{
         add("implicit");
+        add("password");
```
Comment (superafroman) [inline src/main/java/uk/gov/cshr/service/security/ClientDetails.java]: Note, I added this back in to make testing easier - much easier to request a password token than complete the implicit flow.

File: src/main/java/uk/gov/cshr/config/SecurityConfig.java [hunk 2]
```diff
@@ -73,14 +70,13 @@ protected void configure(HttpSecurity http) throws Exception {
         http
                 .antMatcher("/**")
                 .authorizeRequests()
-                    .requestMatchers(forPortAndPath(managementPort, "/", "/login", "/webjars/**")).permitAll()
+                    .requestMatchers(forPortAndPath(managementPort,"/", "/login", "/webjars/**")).permitAll()
```
Comment (elxsy) [inline src/main/java/uk/gov/cshr/config/SecurityConfig.java]: STYLE - removed space

File: src/test/resources/application.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 
 server:
   port: 8080
-  managementPort: 8081
+  managementPort: 8080
```
Comment (elxsy) [inline src/test/resources/application.yml]: INFO - both interfaces on the same port now?

File: src/main/java/uk/gov/cshr/config/SecurityConfig.java (hunk 2 above)
Comment (superafroman) [reply src/main/java/uk/gov/cshr/config/SecurityConfig.java]: Sorted

**Pull request merged**